proto/addressmetadata.proto

```diff
@@ -1,39 +1,74 @@
 syntax = "proto3";
 package models;
 
-message Header {
-    string Name = 1;
-    string Value = 2;
+// Specifies the revocation method
+enum RevocationMethod {
+    /* If any of the pubkeys given in commitment_data[..n-1] signs
+    data commitment_data[n] then revocation is granted. */
+    BackupPubkey = 0;
+}
+
+// Message sent to server to revoke a key
+message Revocation {
+    // The revocation method
+    RevocationMethod method = 1;
+    // Data associated with the reveal phase
+    repeated bytes reveal_data = 2;
```

Comment:
Perhaps this is too generic.

Reply:
repeated should go away.  You should just unmarshal the data into the specific type.

Reply:
Sounds a better idea.